evalml/data_checks/sparsity_data_check.py

```diff
@@ -0,0 +1,90 @@
+from evalml.data_checks import (
+    DataCheck,
+    DataCheckMessageCode,
+    DataCheckWarning
+)
+from evalml.problem_types import ProblemTypes, handle_problem_types
+from evalml.utils.gen_utils import (
+    _convert_to_woodwork_structure,
+    _convert_woodwork_types_wrapper
+)
+
+warning_too_unique = "Input columns ({}) for {} problem type are too sparse."
+
+
+class SparsityDataCheck(DataCheck):
+    """Checks if there are any columns with sparsely populated values in the input."""
+
+    def __init__(self, problem_type, threshold, unique_count_threshold=10):
```

Comment:
Just a thought, but would it be useful to have an arg that could take in a float to represent the minimum percent of the total data for sparsity? Ie, if we pass it 0.1, it could mean that all values have to appear 10% or more in the column for it to be considered non-sparse. Thinking in terms of really large versus smaller datasets, having a percentage might be more useful than just an int. But this seems fine for now!

Some thoughts that I had.

Reply:
I agree, and I think this is a good idea.  I will file an issue for it as I think it makes way more sense than defining a discrete value count threshold.  But for this PR, I just wanted to directly mimic PS' implementation.

Reply:
Issue #1801 filed so that we can come back to this.